components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/SecretManager.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.com).
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.secret.mgt.core;
+
+import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementException;
+import org.wso2.carbon.identity.secret.mgt.core.model.Secret;
+import org.wso2.carbon.identity.secret.mgt.core.model.Secrets;
+
+/**
+ * Secret manager service interface.
+ */
+public interface SecretManager {
+
+    /**
+     * This API is used to create the given secret.
+     *
+     * @param secret The {@link Secret}.
+     * @return 201 created. Returns {@link Secret} created.
+     * @throws SecretManagementException Secret management exception.
+     */
+    Secret addSecret(Secret secret) throws SecretManagementException;
```

Comment:
Will the value provided inside the Secret object is already encrypted one>

Reply:
yes, send the secret with encrypted secret value to this method

Reply:
So, where will be the encryption part happens? the caller should encrypt and send it over the REST API ?